web/themes/custom/cloudy/templates/layout/page.html.twig

```diff
@@ -34,62 +34,87 @@
-</a>
-{# TODO: The skip to content here is a duplicate of the one in html.html.twig #}
-<header id="header" class="header" aria-label="{{ 'Site header'|t}}" role="banner">
-  <nav class="navbar navbar-light navbar-expand-xl" id="navbar-main">
-    <div class="container">
-      {{ page.header }}
-      {% if page.primary_menu or page.header_form %}
-        <button class="navbar-toggler navbar-toggler-right border-0" type="button" data-toggle="collapse" data-target="#CollapsingNavbar" aria-controls="CollapsingNavbar" aria-expanded="false" aria-label="{{ 'Toggle navigation'|t }}">{{ 'Menu'|t }}</button>
-        <div class="collapse navbar-collapse justify-content-end h-100" id="CollapsingNavbar">
-          {{ page.primary_menu }}
-          {% set header_form = page.header_form|render %}
-          {% if header_form %}
-            <div class="form-inline navbar-form">
-              {{ page.header_form }}
-            </div>
+{% block global_admin %}
+  {{ page.global_admin }}
+{% endblock %}
+
+{% block pre_header %}
+  {{ page.pre_header }}
+{% endblock %}
+
+{% block header %}
+  <header id="header" class="header" aria-label="{{ 'Site header'|t}}" role="banner">
+    <nav class="navbar navbar-light navbar-expand-xl" id="navbar-main">
+      <div class="container">
+        {{ page.header }}
+        {% if page.primary_menu or page.header_form %}
+          <button class="navbar-toggler navbar-toggler-right border-0" type="button" data-toggle="collapse" data-target="#CollapsingNavbar" aria-controls="CollapsingNavbar" aria-expanded="false" aria-label="{{ 'Toggle navigation'|t }}">{{ 'Menu'|t }}</button>
+          <div class="collapse navbar-collapse justify-content-end h-100" id="CollapsingNavbar">
+            {{ page.primary_menu }}
+            {% set header_form = page.header_form|render %}
+            {% if header_form %}
+              <div class="form-inline navbar-form">
+                {{ page.header_form }}
+              </div>
+            {% endif %}
+          </div>
+        {% endif %}
+      </div>
+    </nav>
+  </header>
+{% endblock %}
+
+{% block tabs %}
+  {{ page.tabs }}
+{% endblock %}
+
+{% block messages %}
+  {{ page.messages }}
+{% endblock %}
+
+{% block pre_content %}
+  {{ page.pre_content }}
+{% endblock %}
+
+<main id="main-content" role="main" tabindex="-1" data-swiftype-index="true">
+  <section>
+    {% block main_content %}
+      <div class="main-content {{ not is_front ? 'container' }}">
+        <div class="{{ not is_front ? 'row' }}">
+          {% if page.menu %}
+            <aside class="sidebar-left col-sm-12 col-lg-3">
+              {{ page.menu }}
+            </aside>
+          {% endif %}
+          <div class="page-content {{ page.tools ? 'col' }}">
+            {% block page_content %}
+              {{ page.content }}
+            {% endblock %}
+          </div>
+          {% if page.tools %}
+            <aside class="sidebar-right col-sm-12 col-lg-3">
```

Comment:
This doesn't work. You have to put the rendering code within the region or it will print the region even if that region has no content. Refactor so the div and classes are within the regions.